TombEngine/Objects/TR3/Entity/tr3_shiva.cpp

```diff
@@ -463,21 +464,24 @@ namespace TEN::Entities::Creatures::TR3
 
 	void ShivaHit(ItemInfo& target, ItemInfo& source, std::optional<GameVector> pos, int damage, bool isExplosive, int jointIndex)
 	{
-		const auto& player = *GetLaraInfo(&source);
-		const auto& object = Objects[target.ObjectNumber];
-
-		// If guarded, ricochet without damage.
-		if ((target.Animation.ActiveState == SHIVA_STATE_WALK_FORWARD_GUARDING ||
-			target.Animation.ActiveState == SHIVA_STATE_GUARD_IDLE) && pos.has_value())
-		{
-			SoundEffect(SFX_TR4_BADDY_SWORD_RICOCHET, &target.Pose);
-			TriggerRicochetSpark(*pos, source.Pose.Orientation.y, 3, 0);
-		}
-		// Do basic hit effect.
-		else if (object.hitEffect == HitEffect::Blood && pos.has_value())
+		if (pos.has_value())
```

Comment:
You can invert the condition and do early exit instead of doing another block level.

Reply:
Done